src/main/java/seedu/exchangecoursemapper/storage/CourseRepository.java

```diff
@@ -0,0 +1,76 @@
+package seedu.exchangecoursemapper.storage;
+
+import seedu.exchangecoursemapper.courses.Course;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static seedu.exchangecoursemapper.constants.Logs.COURSE_SIZE;
+import static seedu.exchangecoursemapper.constants.Logs.COURSE_ENTRY;
+import static seedu.exchangecoursemapper.constants.Logs.INVALID_DELETE_INDEX;
+import static seedu.exchangecoursemapper.constants.Logs.DELETE_INDEX;
+import static seedu.exchangecoursemapper.constants.Messages.INDEX_OUT_OF_BOUNDS;
+
+public class CourseRepository {
+
+    public static final String MYLIST_FILE_PATH = "./data/myList.json";
+    private static final Logger logger = Logger.getLogger(CourseRepository.class.getName());
+    private final FileHandler fileHandler;
+    private final DataIntegrityChecker dataIntegrityChecker;
+
+    public CourseRepository() {
+        this.fileHandler = new FileHandler(MYLIST_FILE_PATH);
+        this.dataIntegrityChecker = new DataIntegrityChecker();
+    }
+
+    public CourseRepository(String filePath) {
+        this.fileHandler = new FileHandler(filePath);
+        this.dataIntegrityChecker = new DataIntegrityChecker();
+    }
+
+    public boolean isFileValid() {
+        List<String> lines = fileHandler.readAllLines();
+        return dataIntegrityChecker.validateFileIntegrity(lines);
+    }
+
+    public List<Course> loadAllCourses() {
+        List<Course> courses = new ArrayList<>();
+        List<String> lines = fileHandler.readAllLines();
+        for (String line : lines) {
```

Comment:
Good catch! I declared `courses` and `lines` as `List` rather than `ArrayList` because I only use the basic list functionality here, without requiring any specific `ArrayList` behavior. I think this will make the intent of the code clearer and ensures that only list operations are used.